

<!DOCTYPE html>
<!--[if IE 6]>
<html id="ie6" lang="en-CA">
<![endif]-->
<!--[if IE 7]>
<html id="ie7" lang="en-CA">
<![endif]-->
<!--[if IE 8]>
<html id="ie8" lang="en-CA">
<![endif]-->
<!--[if !(IE 6) | !(IE 7) | !(IE 8)  ]><!-->
<html lang="en-CA">
<!--<![endif]-->
<head>
	<meta charset="UTF-8" />
			
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<link rel="pingback" href="http://www.computationalgenomics.ca/xmlrpc.php" />

		<!--[if lt IE 9]>
	<script src="http://www.computationalgenomics.ca/wp-content/themes/Divi/js/html5.js" type="text/javascript"></script>
	<![endif]-->



	<script type="text/javascript">
		document.documentElement.className = 'js';
	</script>

	<script>var et_site_url='http://www.computationalgenomics.ca';var et_post_id='2713';function et_core_page_resource_fallback(a,b){"undefined"===typeof b&&(b=a.sheet.cssRules&&0===a.sheet.cssRules.length);b&&(a.onerror=null,a.onload=null,a.href?a.href=et_site_url+"/?et_core_page_resource="+a.id+et_post_id:a.src&&(a.src=et_site_url+"/?et_core_page_resource="+a.id+et_post_id))}
</script><title>Cvmfs Modules_beta2 - Canadian Centre for Computational Genomics</title>
<script>
  	jvcf7_loading_url= "http://www.computationalgenomics.ca/wp-content/plugins/contact-form-7/images/ajax-loader.gif";
    jvcf7_invalid_field_design = "theme_1";
	jvcf7_show_label_error = "errorMsgshow";
  	</script><link rel='dns-prefetch' href='//maxcdn.bootstrapcdn.com' />
<link rel='dns-prefetch' href='//fonts.googleapis.com' />
<link rel='dns-prefetch' href='//s.w.org' />
<link rel="alternate" type="application/rss+xml" title="Canadian Centre for Computational Genomics &raquo; Feed" href="http://www.computationalgenomics.ca/feed/" />
<link rel="alternate" type="application/rss+xml" title="Canadian Centre for Computational Genomics &raquo; Comments Feed" href="http://www.computationalgenomics.ca/comments/feed/" />
		<script type="text/javascript">
			window._wpemojiSettings = {"baseUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2.3\/72x72\/","ext":".png","svgUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2.3\/svg\/","svgExt":".svg","source":{"concatemoji":"http:\/\/www.computationalgenomics.ca\/wp-includes\/js\/wp-emoji-release.min.js?ver=4.8.2"}};
			!function(a,b,c){function d(a){var b,c,d,e,f=String.fromCharCode;if(!k||!k.fillText)return!1;switch(k.clearRect(0,0,j.width,j.height),k.textBaseline="top",k.font="600 32px Arial",a){case"flag":return k.fillText(f(55356,56826,55356,56819),0,0),b=j.toDataURL(),k.clearRect(0,0,j.width,j.height),k.fillText(f(55356,56826,8203,55356,56819),0,0),c=j.toDataURL(),b!==c&&(k.clearRect(0,0,j.width,j.height),k.fillText(f(55356,57332,56128,56423,56128,56418,56128,56421,56128,56430,56128,56423,56128,56447),0,0),b=j.toDataURL(),k.clearRect(0,0,j.width,j.height),k.fillText(f(55356,57332,8203,56128,56423,8203,56128,56418,8203,56128,56421,8203,56128,56430,8203,56128,56423,8203,56128,56447),0,0),c=j.toDataURL(),b!==c);case"emoji4":return k.fillText(f(55358,56794,8205,9794,65039),0,0),d=j.toDataURL(),k.clearRect(0,0,j.width,j.height),k.fillText(f(55358,56794,8203,9794,65039),0,0),e=j.toDataURL(),d!==e}return!1}function e(a){var c=b.createElement("script");c.src=a,c.defer=c.type="text/javascript",b.getElementsByTagName("head")[0].appendChild(c)}var f,g,h,i,j=b.createElement("canvas"),k=j.getContext&&j.getContext("2d");for(i=Array("flag","emoji4"),c.supports={everything:!0,everythingExceptFlag:!0},h=0;h<i.length;h++)c.supports[i[h]]=d(i[h]),c.supports.everything=c.supports.everything&&c.supports[i[h]],"flag"!==i[h]&&(c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&c.supports[i[h]]);c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&!c.supports.flag,c.DOMReady=!1,c.readyCallback=function(){c.DOMReady=!0},c.supports.everything||(g=function(){c.readyCallback()},b.addEventListener?(b.addEventListener("DOMContentLoaded",g,!1),a.addEventListener("load",g,!1)):(a.attachEvent("onload",g),b.attachEvent("onreadystatechange",function(){"complete"===b.readyState&&c.readyCallback()})),f=c.source||{},f.concatemoji?e(f.concatemoji):f.wpemoji&&f.twemoji&&(e(f.twemoji),e(f.wpemoji)))}(window,document,window._wpemojiSettings);
		</script>
		<meta content="Divi-Child v.2.6.4.4" name="generator"/><style type="text/css">
img.wp-smiley,
img.emoji {
	display: inline !important;
	border: none !important;
	box-shadow: none !important;
	height: 1em !important;
	width: 1em !important;
	margin: 0 .07em !important;
	vertical-align: -0.1em !important;
	background: none !important;
	padding: 0 !important;
}
</style>
<link rel='stylesheet' id='contact-form-7-group-css' href='http://www.computationalgenomics.ca/wp-content/plugins/bwp-minify/min/?f=wp-content/plugins/contact-form-7/includes/css/styles.css,wp-content/plugins/display-medium-posts/public/css/display-medium-posts-public.css,wp-content/plugins/display-medium-posts/public/vendors/owl-carousel/owl.carousel.css,wp-content/plugins/display-medium-posts/public/vendors/owl-carousel/owl.theme.css,wp-content/plugins/jquery-validation-for-contact-form-7/css/jvcf7_validate.css,wp-content/plugins/rotatingtweets/css/style.css,wp-content/plugins/svg-support/css/svgs-attachment.css,wp-content/themes/Divi/style.css,wp-content/themes/Divi-child/style.css,wp-content/themes/Divi/epanel/shortcodes/css/shortcodes.css' type='text/css' media='all' />
<link rel='stylesheet' id='load-fa-css'  href='https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css?ver=4.8.2' type='text/css' media='all' />
<link rel='stylesheet' id='divi-fonts-css'  href='http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800&#038;subset=latin,latin-ext' type='text/css' media='all' />
<link rel='stylesheet' id='et-gf-open-sans-css'  href='http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800&#038;subset=latin,cyrillic-ext,greek-ext,greek,vietnamese,latin-ext,cyrillic' type='text/css' media='all' />
<link rel='stylesheet' id='et-shortcodes-responsive-css-group-css' href='http://www.computationalgenomics.ca/wp-content/plugins/bwp-minify/min/?f=wp-content/themes/Divi/epanel/shortcodes/css/shortcodes_responsive.css,wp-content/themes/Divi/includes/builder/styles/magnific_popup.css' type='text/css' media='all' />
<script type='text/javascript' src='http://www.computationalgenomics.ca/wp-includes/js/jquery/jquery.js?ver=1.12.4'></script>
<script type='text/javascript' src='http://www.computationalgenomics.ca/wp-includes/js/jquery/jquery-migrate.min.js?ver=1.4.1'></script>
<script type='text/javascript' src='http://www.computationalgenomics.ca/wp-content/plugins/bwp-minify/min/?f=wp-content/plugins/rotatingtweets/js/jquery.cycle2.renamed.js,wp-content/plugins/rotatingtweets/js/jquery.cycle2.scrollVert.renamed.js,wp-content/plugins/rotatingtweets/js/jquery.cycle2.carousel.renamed.js,wp-content/plugins/rotatingtweets/js/rotatingtweets_v2.js,wp-content/plugins/display-medium-posts/public/vendors/owl-carousel/owl.carousel.js'></script>
<link rel='https://api.w.org/' href='http://www.computationalgenomics.ca/wp-json/' />
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="http://www.computationalgenomics.ca/xmlrpc.php?rsd" />
<link rel="wlwmanifest" type="application/wlwmanifest+xml" href="http://www.computationalgenomics.ca/wp-includes/wlwmanifest.xml" /> 
<meta name="generator" content="WordPress 4.8.2" />
<link rel="canonical" href="http://www.computationalgenomics.ca/cvmfs-modules/" />
<link rel='shortlink' href='http://www.computationalgenomics.ca/?p=2713' />
<link rel="alternate" type="application/json+oembed" href="http://www.computationalgenomics.ca/wp-json/oembed/1.0/embed?url=http%3A%2F%2Fwww.computationalgenomics.ca%2Fcvmfs-modules%2F" />
<link rel="alternate" type="text/xml+oembed" href="http://www.computationalgenomics.ca/wp-json/oembed/1.0/embed?url=http%3A%2F%2Fwww.computationalgenomics.ca%2Fcvmfs-modules%2F&#038;format=xml" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />		<style id="theme-customizer-css">
							body { color: #494d53; }
								h1, h2, h3, h4, h5, h6 { color: #0074d9; }
										.woocommerce #respond input#submit, .woocommerce-page #respond input#submit, .woocommerce #content input.button, .woocommerce-page #content input.button, .woocommerce-message, .woocommerce-error, .woocommerce-info { background: #0074d9 !important; }
			#et_search_icon:hover, .mobile_menu_bar:before, .mobile_menu_bar:after, .et_toggle_slide_menu:after, .et-social-icon a:hover, .et_pb_sum, .et_pb_pricing li a, .et_pb_pricing_table_button, .et_overlay:before, .entry-summary p.price ins, .woocommerce div.product span.price, .woocommerce-page div.product span.price, .woocommerce #content div.product span.price, .woocommerce-page #content div.product span.price, .woocommerce div.product p.price, .woocommerce-page div.product p.price, .woocommerce #content div.product p.price, .woocommerce-page #content div.product p.price, .et_pb_member_social_links a:hover, .woocommerce .star-rating span:before, .woocommerce-page .star-rating span:before, .et_pb_widget li a:hover, .et_pb_filterable_portfolio .et_pb_portfolio_filters li a.active, .et_pb_filterable_portfolio .et_pb_portofolio_pagination ul li a.active, .et_pb_gallery .et_pb_gallery_pagination ul li a.active, .wp-pagenavi span.current, .wp-pagenavi a:hover, .nav-single a, .posted_in a { color: #0074d9; }
			.et_pb_contact_submit, .et_password_protected_form .et_submit_button, .et_pb_bg_layout_light .et_pb_newsletter_button, .comment-reply-link, .form-submit .et_pb_button, .et_pb_bg_layout_light .et_pb_promo_button, .et_pb_bg_layout_light .et_pb_more_button, .woocommerce a.button.alt, .woocommerce-page a.button.alt, .woocommerce button.button.alt, .woocommerce-page button.button.alt, .woocommerce input.button.alt, .woocommerce-page input.button.alt, .woocommerce #respond input#submit.alt, .woocommerce-page #respond input#submit.alt, .woocommerce #content input.button.alt, .woocommerce-page #content input.button.alt, .woocommerce a.button, .woocommerce-page a.button, .woocommerce button.button, .woocommerce-page button.button, .woocommerce input.button, .woocommerce-page input.button { color: #0074d9; }
			.footer-widget h4 { color: #0074d9; }
			.et-search-form, .nav li ul, .et_mobile_menu, .footer-widget li:before, .et_pb_pricing li:before, blockquote { border-color: #0074d9; }
			.et_pb_counter_amount, .et_pb_featured_table .et_pb_pricing_heading, .et_quote_content, .et_link_content, .et_audio_content, .et_pb_post_slider.et_pb_bg_layout_dark, .et_slide_in_menu_container { background-color: #0074d9; }
									a { color: #0074d9; }
									#main-header .nav li ul { background-color: #f9f9f9; }
									#top-header, #et-secondary-nav li ul { background-color: #555555; }
																
		
					#top-menu li.current-menu-ancestor > a, #top-menu li.current-menu-item > a,
			.et_color_scheme_red #top-menu li.current-menu-ancestor > a, .et_color_scheme_red #top-menu li.current-menu-item > a,
			.et_color_scheme_pink #top-menu li.current-menu-ancestor > a, .et_color_scheme_pink #top-menu li.current-menu-item > a,
			.et_color_scheme_orange #top-menu li.current-menu-ancestor > a, .et_color_scheme_orange #top-menu li.current-menu-item > a,
			.et_color_scheme_green #top-menu li.current-menu-ancestor > a, .et_color_scheme_green #top-menu li.current-menu-item > a { color: #0074d9; }
							#main-footer { background-color: #333333; }
											#main-footer .footer-widget h4 { color: #4397df; }
							.footer-widget li:before { border-color: #4397df; }
															body .et_pb_button,
			.woocommerce a.button.alt, .woocommerce-page a.button.alt, .woocommerce button.button.alt, .woocommerce-page button.button.alt, .woocommerce input.button.alt, .woocommerce-page input.button.alt, .woocommerce #respond input#submit.alt, .woocommerce-page #respond input#submit.alt, .woocommerce #content input.button.alt, .woocommerce-page #content input.button.alt,
			.woocommerce a.button, .woocommerce-page a.button, .woocommerce button.button, .woocommerce-page button.button, .woocommerce input.button, .woocommerce-page input.button, .woocommerce #respond input#submit, .woocommerce-page #respond input#submit, .woocommerce #content input.button, .woocommerce-page #content input.button, .woocommerce-message a.button.wc-forward
			{
													background: #ffffff;
																	border-color: #0074d9;
																					letter-spacing: 1px;
							}
			body.et_pb_button_helper_class .et_pb_button,
			.woocommerce.et_pb_button_helper_class a.button.alt, .woocommerce-page.et_pb_button_helper_class a.button.alt, .woocommerce.et_pb_button_helper_class button.button.alt, .woocommerce-page.et_pb_button_helper_class button.button.alt, .woocommerce.et_pb_button_helper_class input.button.alt, .woocommerce-page.et_pb_button_helper_class input.button.alt, .woocommerce.et_pb_button_helper_class #respond input#submit.alt, .woocommerce-page.et_pb_button_helper_class #respond input#submit.alt, .woocommerce.et_pb_button_helper_class #content input.button.alt, .woocommerce-page.et_pb_button_helper_class #content input.button.alt,
			.woocommerce.et_pb_button_helper_class a.button, .woocommerce-page.et_pb_button_helper_class a.button, .woocommerce.et_pb_button_helper_class button.button, .woocommerce-page.et_pb_button_helper_class button.button, .woocommerce.et_pb_button_helper_class input.button, .woocommerce-page.et_pb_button_helper_class input.button, .woocommerce.et_pb_button_helper_class #respond input#submit, .woocommerce-page.et_pb_button_helper_class #respond input#submit, .woocommerce.et_pb_button_helper_class #content input.button, .woocommerce-page.et_pb_button_helper_class #content input.button {
									color: #0074d9 !important;
							}
							body .et_pb_button:after,
			.woocommerce a.button.alt:after, .woocommerce-page a.button.alt:after, .woocommerce button.button.alt:after, .woocommerce-page button.button.alt:after, .woocommerce input.button.alt:after, .woocommerce-page input.button.alt:after, .woocommerce #respond input#submit.alt:after, .woocommerce-page #respond input#submit.alt:after, .woocommerce #content input.button.alt:after, .woocommerce-page #content input.button.alt:after,
			.woocommerce a.button:after, .woocommerce-page a.button:after, .woocommerce button.button:after, .woocommerce-page button.button:after, .woocommerce input.button:after, .woocommerce-page input.button:after, .woocommerce #respond input#submit:after, .woocommerce-page #respond input#submit:after, .woocommerce #content input.button:after, .woocommerce-page #content input.button:after
			{
															content: '';
										font-size: 20px;
											}
							body .et_pb_button:hover,
			.woocommerce a.button.alt:hover, .woocommerce-page a.button.alt:hover, .woocommerce button.button.alt:hover, .woocommerce-page button.button.alt:hover, .woocommerce input.button.alt:hover, .woocommerce-page input.button.alt:hover, .woocommerce #respond input#submit.alt:hover, .woocommerce-page #respond input#submit.alt:hover, .woocommerce #content input.button.alt:hover, .woocommerce-page #content input.button.alt:hover,
			.woocommerce a.button:hover, .woocommerce-page a.button:hover, .woocommerce button.button:hover, .woocommerce-page button.button:hover, .woocommerce input.button:hover, .woocommerce-page input.button:hover, .woocommerce #respond input#submit:hover, .woocommerce-page #respond input#submit:hover, .woocommerce #content input.button:hover, .woocommerce-page #content input.button:hover
			{
									 color: #f5f5f5 !important;
													background: #0074d9 !important;
													border-color: #ffffff !important;
															}
		
		
																														
		@media only screen and ( min-width: 981px ) {
												.et_pb_section { padding: 1% 0; }
					.et_pb_section.et_pb_section_first { padding-top: inherit; }
					.et_pb_fullwidth_section { padding: 0; }
													.et_pb_row { padding: 3% 0; }
																		.footer-widget h4 { font-size: 16px; }
										.et_header_style_left #et-top-navigation, .et_header_style_split #et-top-navigation  { padding: 41px 0 0 0; }
				.et_header_style_left #et-top-navigation nav > ul > li > a, .et_header_style_split #et-top-navigation nav > ul > li > a { padding-bottom: 41px; }
				.et_header_style_split .centered-inline-logo-wrap { width: 81px; margin: -81px 0; }
				.et_header_style_split .centered-inline-logo-wrap #logo { max-height: 81px; }
				.et_pb_svg_logo.et_header_style_split .centered-inline-logo-wrap #logo { height: 81px; }
				.et_header_style_centered #top-menu > li > a { padding-bottom: 15px; }
				.et_header_style_slide #et-top-navigation, .et_header_style_fullscreen #et-top-navigation { padding: 32px 0 32px 0 !important; }
									.et_header_style_centered #main-header .logo_container { height: 81px; }
														#logo { max-height: 71%; }
				.et_pb_svg_logo #logo { height: 71%; }
																			.et_header_style_centered.et_hide_primary_logo #main-header:not(.et-fixed-header) .logo_container, .et_header_style_centered.et_hide_fixed_logo #main-header.et-fixed-header .logo_container { height: 14.58px; }
																.et-fixed-header#top-header, .et-fixed-header#top-header #et-secondary-nav li ul { background-color: #555555; }
													.et-fixed-header #top-menu li a { font-size: 24px; }
													.et-fixed-header #top-menu li.current-menu-ancestor > a,
				.et-fixed-header #top-menu li.current-menu-item > a { color: #0074d9 !important; }
						
					}
					@media only screen and ( min-width: 1350px) {
				.et_pb_row { padding: 40px 0; }
				.et_pb_section { padding: 13px 0; }
				.single.et_pb_pagebuilder_layout.et_full_width_page .et_post_meta_wrapper { padding-top: 121px; }
				.et_pb_section.et_pb_section_first { padding-top: inherit; }
				.et_pb_fullwidth_section { padding: 0; }
			}
		
		@media only screen and ( max-width: 980px ) {
																				}
		@media only screen and ( max-width: 767px ) {
														}
	</style>

	
					<style class="et_heading_font">
				h1, h2, h3, h4, h5, h6 {
					font-family: 'Open Sans', Helvetica, Arial, Lucida, sans-serif;				}
				</style>
							<style class="et_body_font">
				body, input, textarea, select {
					font-family: 'Open Sans', Helvetica, Arial, Lucida, sans-serif; font-weight: 300;				}
				</style>
							<style class="et_all_buttons_font">
				.et_pb_button {
					font-family: 'Open Sans', Helvetica, Arial, Lucida, sans-serif;				}
				</style>
							<style class="et_primary_nav_font">
				#main-header,
				#et-top-navigation {
					font-family: 'Open Sans', Helvetica, Arial, Lucida, sans-serif;				}
				</style>
							<style class="et_secondary_nav_font">
				#top-header .container{
					font-family: 'Open Sans', Helvetica, Arial, Lucida, sans-serif;				}
				</style>
			
	
	<style id="module-customizer-css">
		.et_pb_subscribe { padding: 13px; }
.et_pb_contact p input, .et_pb_contact p textarea { padding: 10px; }
.et_pb_fullwidth_section .et_pb_slide_description .et_pb_slide_title { font-size: 40px; }
.et_pb_fullwidth_section .et_pb_slide_content { font-size: 16px; }
.et_pb_fullwidth_section .et_pb_slide_description { padding-top: 7%; padding-bottom: 7% }
	</style>

	<link rel="shortcut icon" href="http://www.computationalgenomics.ca/wp-content/uploads/2016/02/globe_small.png" /><meta name="twitter:card" content="summary" /><meta name="twitter:domain" content="Canadian Centre for Computational Genomics" /><meta name="twitter:title" content="Cvmfs Modules_beta2 - Canadian Centre for Computational Genomics" /><meta property="og:site_name" content="Canadian Centre for Computational Genomics" /><meta property="og:url" content="http://www.computationalgenomics.ca/cvmfs-modules/" /><meta property="og:type" content="article" /><meta property="og:title" content="Cvmfs Modules_beta2 - Canadian Centre for Computational Genomics" /><meta name="title" content="Cvmfs Modules_beta2 - Canadian Centre for Computational Genomics" /><link rel="icon" href="http://www.computationalgenomics.ca/wp-content/uploads/2016/04/cropped-globe2-32x32.png" sizes="32x32" />
<link rel="icon" href="http://www.computationalgenomics.ca/wp-content/uploads/2016/04/cropped-globe2-192x192.png" sizes="192x192" />
<link rel="apple-touch-icon-precomposed" href="http://www.computationalgenomics.ca/wp-content/uploads/2016/04/cropped-globe2-180x180.png" />
<meta name="msapplication-TileImage" content="http://www.computationalgenomics.ca/wp-content/uploads/2016/04/cropped-globe2-270x270.png" />

<!-- BEGIN ExactMetrics v5.3.8 Universal Analytics - https://exactmetrics.com/ -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-61400807-1', 'auto');
  ga('send', 'pageview');
</script>
<!-- END ExactMetrics Universal Analytics -->
		<style type="text/css" id="wp-custom-css">
			a:hover{
text-decoration:underline;
}
a.et_pb_button{
text-decoration:none;
}
.cycle-carousel-wrap .rotatingtweet {
margin-bottom: 26px !important;
border-bottom:1px solid #F2f2f2;
}
.rotatingtweets {

    margin-top: 26px !important;
}
.wkshopEvents ul.eo-events li{
list-style:none !important;
margin:20px 0 20px 0;
}
.wkshopEvents ul.eo-events{
padding:0;
}
/*contact form 7 */
.wpcf7-text, .wpcf7-textarea, .wpcf7-captchar, .wpcf7-file{
background-color: #fff!important;
border: none !important;
width: 100% !important;
-moz-border-radius: 0 !important;
-webkit-border-radius: 0 !important;
border-radius: 0 !important;
font-size: 14px;
color: #333!important;
padding: 16px !important;
-moz-box-sizing: border-box;
-webkit-box-sizing: border-box;
box-sizing: border-box;
}
.wpcf7-file{
    padding: 14px !important;
}
.et_pb_widget{
width:100%
}
/**Footer custom styles for blurb*/
.footerblurb.et_pb_blurb_6.et_pb_blurb h4, .footerblurb.et_pb_blurb_6.et_pb_blurb h4 a {
    color: #ffffff !important;
    font-size: 12px !important;
}
.footerblurb.et_pb_blurb_6.et_pb_blurb h4 a:hover{
   color: #4397df!important;
} 
.footerblurb.et_pb_blurb_container {
    padding-left: 6px;
}

.footerblurb.et_pb_blurb_6 .et-pb-icon {
    font-size: 14px;
}

/*Card flip effect styles*/ 
/* entire container, keeps perspective */
.flip-container {
	perspective: 700;
	transform-style: preserve-3d;
}
	/*  UPDATED! flip the pane when hovered */
	.flip-container:hover .back {
		transform: rotateY(0deg);
	}
	.flip-container:hover .front {
	    transform: rotateY(180deg);
	}

.flip-container, .front, .back {
/*		height: 100%;
		width:100%;
	*/	
		width:250px;
height:250px;
}

/* flip speed goes here */
.flipper {
	transition: 0.9s;
	transform-style: preserve-3d;
	/*position: relative;*/
	/*width:200px;*/
}

/* hide back of pane during swap */
.front, .back {
	backface-visibility: hidden;
	transition: 0.9s;
	transform-style: preserve-3d;
padding:20px;
	position: absolute;
	top: 0;
	left: 0;
}

/*  UPDATED! front pane, placed above back */
.front {
	z-index: 2;
	
	transform: rotateY(0deg);
}

/* back, initially hidden pane */
.back {
	transform: rotateY(-180deg);

}
/*End card flip effect styles*/

/* captcha filed width*/

.et_pb_contact_right p input {
    max-width: 60px;
    padding: 16px;
}

/*Border for Contact dav*/

.wpcf7-form .border-contact {

    padding: 2px;
    border: 1px solid #bbb !important;
    color: #4e4e4e;
    background-color: #fff;
}

/*contact submit button*/

.contact-btn{
    background: #ffffff;
    border-color: #0074d9;
    letter-spacing: 1px;
    color:#0074d9;
   padding: 0.3em 1em !important;
     font-size: 16px;
     font-weight: 400;
  margin-top:15px;
}

.contact-btn:hover{
  background: #0074d9;
  color:#ffffff;
}


/*feedback on Contact form*/

div.wpcf7-mail-sent-ok {
    border: 2px solid #fefefe;
}

/*feedback on Contact form*/


/*spacing for contact form*/

.label {
padding-top: 80px !important;
}


/*spacing for contact form*/

form p {

padding-bottom: 0px;

}

/* service form----*/





wpcf7-form-control wpcf7-text wpcf7-validates-as-required border-contact {
height: 30px;

}		</style>
	
	<link rel="stylesheet" type="text/css" media="screen" href="Canadian Centre for Computational Genomics" />
</head>
<body class="page-template-default page page-id-2713 et_divi_builder et_button_left et_button_icon_visible et_button_no_icon et_button_custom_icon et_pb_button_helper_class et_fullwidth_secondary_nav et_non_fixed_nav et_show_nav et_pb_gutter et_pb_gutters2 et_primary_nav_dropdown_animation_fade et_secondary_nav_dropdown_animation_fade et_pb_footer_columns3 et_header_style_left et_pb_pagebuilder_layout et_right_sidebar et_divi_theme unknown">
	<div id="page-container">

	
	
		<header id="main-header" data-height-onload="81">
			<div class="container clearfix et_menu_container">
							<div class="logo_container">
					<span class="logo_helper"></span>
					<a href="http://www.computationalgenomics.ca/">
						<img src="http://www.computationalgenomics.ca/wp-content/uploads/2016/02/c3g_Web.png" alt="Canadian Centre for Computational Genomics" id="logo" data-height-percentage="71" />
					</a>
				</div>
				<div id="et-top-navigation" data-height="81" data-fixed-height="40">
											<nav id="top-menu-nav">
						<ul id="top-menu" class="nav"><li id="menu-item-79" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-home menu-item-79"><a href="http://www.computationalgenomics.ca/">Home</a></li>
<li id="menu-item-82" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-has-children menu-item-82"><a href="http://www.computationalgenomics.ca/about/">About</a>
<ul  class="sub-menu">
	<li id="menu-item-824" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-824"><a href="http://www.computationalgenomics.ca/about/">C3G</a></li>
	<li id="menu-item-3916" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-3916"><a href="http://www.computationalgenomics.ca/blog/">Blog</a></li>
	<li id="menu-item-3907" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-3907"><a href="http://www.computationalgenomics.ca/publication/">Publications</a></li>
	<li id="menu-item-102" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-102"><a href="http://www.computationalgenomics.ca/montreal-node/">Montreal Node</a></li>
	<li id="menu-item-101" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-101"><a href="http://www.computationalgenomics.ca/toronto-node/">Toronto Node</a></li>
	<li id="menu-item-100" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-100"><a href="http://www.computationalgenomics.ca/partners/">Partners &#038; Funders</a></li>
	<li id="menu-item-99" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-99"><a href="http://www.computationalgenomics.ca/teams/">Teams</a></li>
	<li id="menu-item-103" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-103"><a href="http://www.computationalgenomics.ca/careers/">Careers</a></li>
	<li id="menu-item-3695" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-3695"><a href="http://www.computationalgenomics.ca/interns/">Interns</a></li>
</ul>
</li>
<li id="menu-item-80" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-80"><a href="http://www.computationalgenomics.ca/services/">Services</a></li>
<li id="menu-item-562" class="menu-item menu-item-type-post_type menu-item-object-page current-menu-ancestor current-menu-parent current_page_parent current_page_ancestor menu-item-has-children menu-item-562"><a href="http://www.computationalgenomics.ca/tools/">Tools</a>
<ul  class="sub-menu">
	<li id="menu-item-3518" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-3518"><a href="http://www.computationalgenomics.ca/online-applications/">Online Applications</a></li>
	<li id="menu-item-2356" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-2356"><a href="http://www.computationalgenomics.ca/tools/">Bioinformatic Tools</a></li>
	<li id="menu-item-2440" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-has-children menu-item-2440"><a href="http://www.computationalgenomics.ca/tutorials/">Tutorials</a>
	<ul  class="sub-menu">
		<li id="menu-item-2909" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2909"><a href="http://www.computationalgenomics.ca/get-a-mammouth-account/">Get a Mammouth Account</a></li>
		<li id="menu-item-2938" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2938"><a href="http://www.computationalgenomics.ca/genpipes-in-the-cloud/">GenPipes in the Cloud</a></li>
	</ul>
</li>
	<li id="menu-item-2781" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2781"><a href="http://www.computationalgenomics.ca/test-dataset/">Test Dataset</a></li>
	<li id="menu-item-2723" class="menu-item menu-item-type-post_type menu-item-object-page current-menu-item page_item page-item-2713 current_page_item menu-item-2723"><a href="http://www.computationalgenomics.ca/cvmfs-modules/">CVMFS Modules</a></li>
	<li id="menu-item-2739" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2739"><a href="http://www.computationalgenomics.ca/cvmfs-genomes/">CVMFS Genomes</a></li>
</ul>
</li>
<li id="menu-item-2668" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-has-children menu-item-2668"><a href="http://www.computationalgenomics.ca/2019-2/">Training</a>
<ul  class="sub-menu">
	<li id="menu-item-3488" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-3488"><a href="http://www.computationalgenomics.ca/2019-2/">2019</a></li>
	<li id="menu-item-2593" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2593"><a href="http://www.computationalgenomics.ca/2018-2/">2018</a></li>
	<li id="menu-item-1937" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-1937"><a href="http://www.computationalgenomics.ca/workshop2017/">2017</a></li>
	<li id="menu-item-1936" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-1936"><a href="http://www.computationalgenomics.ca/2016-2/">2016</a></li>
	<li id="menu-item-1926" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-1926"><a href="http://www.computationalgenomics.ca/2015-2/">2015</a></li>
	<li id="menu-item-1739" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-1739"><a href="http://www.computationalgenomics.ca/past-events/">Past Events</a></li>
</ul>
</li>
<li id="menu-item-2188" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-2188"><a href="http://www.computationalgenomics.ca/open-door/">Open Door</a></li>
<li id="menu-item-1903" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-1903"><a href="http://www.computationalgenomics.ca/help-desk/">Help Desk</a></li>
<li id="menu-item-81" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-81"><a href="http://www.computationalgenomics.ca/contact/">Contact</a></li>
</ul>						</nav>
					
					
					
										<div id="et_top_search">
						<span id="et_search_icon"></span>
					</div>
					
					<div id="et_mobile_nav_menu">
				<div class="mobile_nav closed">
					<span class="select_page">Select Page</span>
					<span class="mobile_menu_bar mobile_menu_bar_toggle"></span>
				</div>
			</div>				</div> <!-- #et-top-navigation -->
			</div> <!-- .container -->
			<div class="et_search_outer">
				<div class="container et_search_form_container">
					<form role="search" method="get" class="et-search-form" action="http://www.computationalgenomics.ca/">
					<input type="search" class="et-search-field" placeholder="Search &hellip;" value="" name="s" title="Search for:" />					</form>
					<span class="et_close_search_field"></span>
				</div>
			</div>
		</header> <!-- #main-header -->

		<div id="et-main-area">

<div id="main-content">


			
				<article id="post-2713" class="post-2713 page type-page status-publish hentry">

				
					<div class="entry-content">
					<div class="et_pb_section et_pb_fullwidth_section  et_pb_section_0 et_section_regular">
				
				
					
					<section class="et_pb_fullwidth_header et_pb_module et_pb_bg_layout_dark et_pb_text_align_center  et_pb_fullwidth_header_0">
				
				<div class="et_pb_fullwidth_header_container center">
					<div class="header-content-container center">
					<div class="header-content">
						
						<h1>CVMFS Modules</h1>
						
						

						
					</div>
				</div>
					
				</div>
				<div class="et_pb_fullwidth_header_overlay"></div>
				<div class="et_pb_fullwidth_header_scroll"></div>
			</section>
				
			</div> <!-- .et_pb_section --><div class="et_pb_section  et_pb_section_1 et_section_regular">
				
				
					
					<div class=" et_pb_row et_pb_row_0">
				
				<div class="et_pb_column et_pb_column_4_4  et_pb_column_0 et_pb_column_empty">
				
				
			</div> <!-- .et_pb_column -->
					
			</div> <!-- .et_pb_row --><div class=" et_pb_row et_pb_row_1">
				
				<div class="et_pb_column et_pb_column_4_4  et_pb_column_1">
				
				<div class="et_pb_text et_pb_module et_pb_bg_layout_light et_pb_text_align_left  et_pb_text_0">
				
<h1>Bioinformatics resources</h1>
<p>&nbsp;</p>
<p>C3G, in partnership with Compute Canada, offers and maintains a large set of bioinformatics resources for the community. Below is a list of software currently deployed on several HPC centres, including Guillimin, Cedar, Graham and Mammouth. Several reference genomes are also <a href="http://www.computationalgenomics.ca/cvmfs-genomes/">available</a>.</p>
<p>&nbsp;</p>
<p>To access these modules, you need to add the following lines to your .bashrc file.</p>
<div class="codebox">
<pre>## GenPipes/MUGQIC genomes and modules
export MUGQIC_INSTALL_HOME=/cvmfs/soft.mugqic/CentOS6
module use $MUGQIC_INSTALL_HOME/modulefiles
</pre>
</div>
<p>&nbsp;</p>
<p>To explore the available tools, you can type:</p>
<div class="codebox">
<pre>module avail mugqic/
</pre>
</div>
<p>To load a module, for example samtools, you can use:</p>
<div class="codebox">
<pre># module add mugqic/&lt;tool&gt;/&lt;version&gt;
module add mugqic/samtools/1.4.1
# Now samtools 1.4.1 is available to use. To check:
samtools
</pre>
</div>
<p>&nbsp;<br />
&nbsp;</p>
<h2>Available Modules</h2>

			</div> <!-- .et_pb_text --><div class="et_pb_code et_pb_module  et_pb_code_0">
				<h5></h5>
<p>
<a href="#anaconda">Anaconda</a>, <a href="#ASCAT">ASCAT</a>, <a href="#aspera-connect">Aspera Connect</a>, <a href="#ATLAS">ATLAS</a>, <a href="#bcftools">BCFtools</a>, <a href="#beagle">Beagle</a>, <a href="#bedtools">bedtools</a>, <a href="#bismark">bismark</a>, <a href="#BisSNP">BisSNP</a>, <a href="#blast">BLAST</a>, <a href="#boost">boost</a>, <a href="#bowtie">Bowtie</a>, <a href="#bowtie2">Bowtie2</a>, <a href="#breakdancer">BbreakDancer</a>, <a href="#butter">butter</a>, <a href="#bvatools">bvatools</a>, <a href="#bwa">BWA</a>, <a href="#bwakit">Bwakit</a>, <a href="#cd-hit">CD-HIT</a>, <a href="#cellranger">Cell Ranger</a>, <a href="#cufflinks">Cufflinks</a>, <a href="#duk">duk</a>, <a href="#ea-utils">ea-utils</a>, <a href="#emboss">emboss</a>, <a href="#EPACTS">EPACTS</a>, <a href="#exonerate">Exonerate</a>, <a href="#fastqc">FastQC</a>, <a href="#fasttree">FastTree</a>, <a href="#fastx">FASTX-Toolkit</a>, <a href="#FLASH">FLASH</a>, <a href="#gcc">gcc</a>, <a href="#gemini">GEMINI</a>, <a href="#gemLibrary">GEM</a>, <a href="#GenomeAnalysisTK">Genome Analysis Toolkit</a>, <a href="#ghostscript">Ghostscript</a>, <a href="#gnuplot">Gnuplot</a>, <a href="#hmmer">HMMER</a>, <a href="#homer">HOMER</a>, <a href="#htslib">HTSlib</a>, <a href="#igv">IGV</a>, <a href="#igvtools">igvtools</a>, <a href="#java">Java</a>, <a href="#jellyfish">JELLYFISH</a>, <a href="#kentUtils">kentUtils</a>, <a href="#kmergenie">KmerGenie</a>, <a href="#KronaTools">KronaTools</a>, <a href="#lapack">LAPACK</a>, <a href="#longranger">Long Ranger</a>, <a href="#MACS">MACS</a>, <a href="#MACS2">MACS2</a>, <a href="#mirdeep2">miRDeep2</a>, <a href="#mpich">mpich</a>, <a href="#mugqic_pipelines">mugqic_pipelines</a>, <a href="#mugqic_R_packages">mugqic_R_packages</a>, <a href="#mugqic_tools">mugqic_tools</a>, <a href="#MUMmer">MUMmer</a>, <a href="#MUSCLE">MUSCLE</a>, <a href="#mutect">MuTect</a>, <a href="#nextclip">NextClip</a>, <a href="#OpenBLAS">OpenBLAS</a>, <a href="#pandoc">Pandoc</a>, <a href="#parallel">parallel</a>, <a href="#pbs-drmaa">pbs-drmaa</a>, <a href="#perl">perl</a>, <a href="#picard">Picard</a>, <a href="#pigz">pigz</a>, <a href="#prinseq-lite">PRINSEQ-lite</a>, <a href="#prodigal">Prodigal</a>, <a href="#python">Python</a>, <a href="#qualimap">qualimap</a>, <a href="#R_Bioconductor">R_Bioconductor</a>, <a href="#ray">Ray</a>, <a href="#rnammer">RNAmmer</a>, <a href="#rnaseqc">RNA-SeQC</a>, <a href="#rsem">RSEM</a>, <a href="#samtools">SAMtools</a>, <a href="#scalpel">Scalpel</a>, <a href="#ShortStack">ShortStack</a>, <a href="#signalp">SignalP</a>, <a href="#smrtanalysis">SMRT-Analysis</a>, <a href="#snap">SNAP</a>, <a href="#snpEff">SnpEff</a>, <a href="#sphinx">Sphinx</a>, <a href="#SPAdes">SPAdes</a>, <a href="#SplAdder">SplAdder</a>, <a href="#star">STAR</a>, <a href="#supernova">Supernova</a>, <a href="#tabix">Tabix</a>, <a href="#tmhmm">TMHMM</a>, <a href="#tools">tools</a>, <a href="#tophat">TopHat</a>, <a href="#TransDecoder">TransDecoder</a>, <a href="#trimmomatic">Trimmomatic</a>, <a href="#trinity">Trinity</a>, <a href="#trinotate">Tinotate</a>, <a href="#ucsc">UCSC tools</a>, <a href="#usearch">USEARCH</a>, <a href="#VarScan">VarScan</a>, <a href="#vcftools">VCFtools</a>, <a href="#verifyBamID">VerifyBamID</a>, <a href="#ViennaRNA">ViennaRNA</a>, <a href="#vsearch">VSEARCH</a>, <a href="#vt">vt</a>, <a href="#weblogo">WebLogo</a>, <a href="#wgs-assembler">Celera Assembler</a></p>
<p><h3>List of Modules</h3>
<p>
<strong><a href="https://www.continuum.io" name="anaconda">Anaconda</a></strong><br />
Anaconda is a freemium open source distribution of the Python and R programming languages for large-scale data processing, predictive analytics, and scientific computing, that aims to simplify package management and deployment.<br />
<em>Versions Available:</em> 2-4.0.0</p>
<p><strong><a href="https://www.crick.ac.uk/peter-van-loo/software/ASCAT" name="ASCAT">ASCAT</a></strong><br />
A tool for accurate dissection of genome-wide allele-specific copy number in tumors.<br />
<em>Versions Available:</em> 2.3</p>
<p><strong><a href="http://downloads.asperasoft.com/connect2/" name="aspera-connect">Aspera Connect</a></strong><br />
High-performance transfer plug-in<br />
<em>Versions Available:</em> 3.3.3</p>
<p><strong><a href="https://sourceforge.net/projects/atlasdb" name="ATLAS">ATLAS</a></strong><br />
A data warehouse for integrative bioinformatics<br />
<em>Versions Available:</em> 3.10.2</p>
<p><strong><a href="https://samtools.github.io/bcftools/bcftools.html" name="bcftools">BCFtools</a></strong><br />
Utilities for variant calling and manipulating VCFs and BCFs<br />
<em>Versions Available:</em> 1.2, 1.3</p>
<p><strong><a href="https://faculty.washington.edu/browning/beagle/beagle.html" name="beagle">Beagle</a></strong><br />
Beagle is a software package that performs genotype calling, genotype phasing, imputation of ungenotyped markers, and identity-by-descent segment detection.<br />
<em>Versions Available:</em> 03May16.862, r13994</p>
<p><strong><a href="https://github.com/arq5x/bedtools2" name="bedtools">bedtools</a></strong><br />
A software suite for the comparison, manipulation and annotation of genomic features in browser extensible data (BED) and general feature format (GFF) format.<br />
<em>Versions Available:</em> 2.17.0, 2.22.1, 2.25.0, 2.26.0</p>
<p><strong><a href="https://github.com/FelixKrueger/Bismark" name="bismark">bismark</a></strong><br />
Bismark is a program to map bisulfite treated sequencing reads to a genome of interest and perform methylation calls in a single step.<br />
<em>Versions Available:</em> 0.16.1, 0.16.3, 0.17.0</p>
<p><strong><a href="http://people.csail.mit.edu/dnaase/bissnp2011/" name="BisSNP">BisSNP</a></strong><br />
A bisulfite space genotyper &amp; methylation caller<br />
<em>Versions Available:</em> 0.82.2</p>
<p><strong><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download" name="blast">BLAST</a></strong><br />
Basic Local Alignment Search Tool<br />
<em>Versions Available:</em> 2.2.29+, 2.3.0+</p>
<p><strong><a href="http://www.boost.org/" name="boost">boost</a></strong><br />
Boost provides free portable peer-reviewed C++ libraries. The emphasis is on portable libraries which work well with the C++ Standard Library.<br />
<em>Versions Available:</em> 1.63.0</p>
<p><strong><a href="http://bowtie-bio.sourceforge.net/index.shtml" name="bowtie">Bowtie</a></strong><br />
An ultrafast, memory-efficient short read aligner.<br />
<em>Versions Available:</em> 1.0.0, 1.1.2, 1.2.0</p>
<p><strong><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" name="bowtie2">Bowtie2</a></strong><br />
Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences.<br />
<em>Versions Available:</em> 2.2.3, 2.2.4, 2.2.9</p>
<p><strong><a href="http://gmt.genome.wustl.edu/packages/breakdancer/" name="breakdancer">BbreakDancer</a></strong><br />
A Perl/C++ package that provides genome-wide detection of structural variants from next generation paired-end sequencing reads.<br />
<em>Versions Available:</em> 1.1_2011_02_21</p>
<p><strong><a href="https://github.com/MikeAxtell/butter" name="butter">butter</a></strong><br />
A wrapper for bowtie to produce small RNA-seq alignments where multimapped small RNAs tend to be placed near regions of confidently high density.<br />
<em>Versions Available:</em> 0.3.3</p>
<p><strong><a href="https://bitbucket.org/mugqic/bvatools" name="bvatools">bvatools</a></strong><br />
BVATools -- Bam and Variant Analysis Tools<br />
<em>Versions Available:</em> 1.3, 1.4, 1.5, 1.6</p>
<p><strong><a href="http://bio-bwa.sourceforge.net/" name="bwa">BWA</a></strong><br />
A software package for mapping low-divergent sequences against a large reference genome, such as the human genome. <br />
<em>Versions Available:</em> 0.7.10, 0.7.12</p>
<p><strong><a href="https://github.com/lh3/bwa/tree/master/bwakit" name="bwakit">Bwakit</a></strong><br />
Bwakit is a self-consistent installation-free package of scripts and precompiled binaries, providing an end-to-end solution to read mapping.<br />
<em>Versions Available:</em> 0.7.12</p>
<p><strong><a href="http://weizhongli-lab.org/cd-hit/" name="cd-hit">CD-HIT</a></strong><br />
CD-HIT is a very widely used program for clustering and comparing protein or nucleotide sequences. <br />
<em>Versions Available:</em> 4.5.4-2011-03-07</p>
<p><strong><a href="http://www.10xgenomics.com/" name="cellranger">Cell Ranger</a></strong><br />
Cell Ranger is a set of analysis pipelines that processes Chromium single cell 3â€™ RNA-seq output to align reads, generate gene-cell matrices and perform clustering and gene expression analysis.<br />
<em>Versions Available:</em> 1.3.0</p>
<p><strong><a href="http://cole-trapnell-lab.github.io/cufflinks/" name="cufflinks">Cufflinks</a></strong><br />
Cufflinks assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples. <br />
<em>Versions Available:</em> 2.2.1</p>
<p><strong><a href="http://duk.sourceforge.net/" name="duk">duk</a></strong><br />
Duk is a fast, accurate,and memory efficent DNA sequence matching tool. It finds whether a query sequence partially or totally matches given reference sequences or not, but it does not give how a query matches a reference sequence. The common application is to group sequencing reads into small manageable chunks for downstream analysis in assessing quality of a sequencing run, which includes contaminant removal (with contaminant sequences known), organelle genome separation, and assembly refinement.<br />
<em>Versions Available:</em> 1.1</p>
<p><strong><a href="https://code.google.com/p/ea-utils/" name="ea-utils">ea-utils</a></strong><br />
A command-line tools for processing biological sequencing data. Barcode demultiplexing, adapter trimming, etc. <br />
<em>Versions Available:</em> 1.1.2-537</p>
<p><strong><a href="http://emboss.sourceforge.net/" name="emboss">emboss</a></strong><br />
EMBOSS is &#039;The European Molecular Biology Open Software Suite&#039;. EMBOSS is a free Open Source software analysis package specially developed for the needs of the molecular biology (e.g. EMBnet) user community.<br />
<em>Versions Available:</em> 6.4.0, 6.6.0</p>
<p><strong><a href="http://genome.sph.umich.edu/wiki/EPACTS" name="EPACTS">EPACTS</a></strong><br />
A versatile software pipeline to perform various statistical tests for identifying genome-wide association from sequence data through a user-friendly interface, both to scientific analysts and to method developers.<br />
<em>Versions Available:</em> 3.2.6</p>
<p><strong><a href="http://www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate" name="exonerate">Exonerate</a></strong><br />
A generic tool for pairwise sequence comparison.<br />
<em>Versions Available:</em> 2.2.0</p>
<p><strong><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" name="fastqc">FastQC</a></strong><br />
A quality control tool for high throughput sequence data.<br />
<em>Versions Available:</em> 0.11.2, 0.11.5</p>
<p><strong><a href="http://www.microbesonline.org/fasttree/" name="fasttree">FastTree</a></strong><br />
FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences.<br />
<em>Versions Available:</em> 2.1.7</p>
<p><strong><a href="http://hannonlab.cshl.edu/fastx_toolkit/" name="fastx">FASTX-Toolkit</a></strong><br />
The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.<br />
<em>Versions Available:</em> 0.0.14</p>
<p><strong><a href="http://ccb.jhu.edu/software/FLASH/" name="FLASH">FLASH</a></strong><br />
FLASH (Fast Length Adjustment of SHort reads) is a very fast and accurate software tool to merge paired-end reads from next-generation sequencing experiments. FLASH is designed to merge pairs of reads when the original DNA fragments are shorter than twice the length of reads. The resulting longer reads can significantly improve genome assemblies.<br />
<em>Versions Available:</em>  1.2.8, 1.2.11</p>
<p><strong><a href="https://gcc.gnu.org/" name="gcc">gcc</a></strong><br />
The GNU Compiler Collection includes front ends for C, C++, Objective-C, Fortran, Java, Ada, and Go, as well as libraries for theselanguages (libstdc++, libgcj,...). GCC was originally written as the compiler for the GNU operating system.<br />
<em>Versions Available:</em> 4.9.3</p>
<p><strong><a href="https://gemini.readthedocs.io/en/latest/ " name="gemini">GEMINI</a></strong><br />
Flexible framework for exploring genetic variation in the context of the wealth of genome annotations available for the human genome.<br />
<em>Versions Available:</em> 0.18.0, 0.18.2, 0.18.3</p>
<p><strong><a href="http://algorithms.cnag.cat/wiki/The_GEM_library" name="gemLibrary">GEM</a></strong><br />
The GEM library strives to be a true &#039;next-generation&#039; tool for handling any kind of sequence data, offering state-of-the-art algorithms and data structures specifically tailored to this demanding task.<br />
<em>Versions Available:</em> v1.315</p>
<p><strong><a href="https://software.broadinstitute.org/gatk/" name="GenomeAnalysisTK">Genome Analysis Toolkit</a></strong><br />
Developed by the Data Science and Data Engineering group at the Broad Institute, the toolkit offers a wide variety of tools with a primary focus on variant discovery and genotyping.<br />
<em>Versions Available:</em> 3.2-2, 3.3-0, 3.5, 3.7</p>
<p><strong><a href="http://ghostscript.com/" name="ghostscript">Ghostscript</a></strong><br />
An interpreter for the PostScript language and for PDF.<br />
<em>Versions Available:</em> &#039;8.70&#039;</p>
<p><strong><a href="http://www.gnuplot.info/ " name="gnuplot">Gnuplot</a></strong><br />
Gnuplot  is a portable command-line driven graphing utility for Linux, OS/2, MS Windows, OSX, VMS, and many other platforms.<br />
<em>Versions Available:</em> 4.6.4, 4.6.6</p>
<p><strong><a href="http://hmmer.org/ " name="hmmer">HMMER</a></strong><br />
HMMER is used for searching sequence databases for sequence homologs,and for making sequence alignments. It implements methods using probabilistic models called profile hidden Markov models (profile HMMs). <br />
<em>Versions Available:</em> 2.3.2, 3.1b1, 3.1b2</p>
<p><strong><a href="http://homer.salk.edu/homer/ngs/ " name="homer">HOMER</a></strong><br />
HOMER offers tools and methods for interpreting Next-gen-Seq experiments.Â  In addition to Genome Browser/UCSC visualization support and peak finding [and motif finding of course], HOMER can help assemble data across multiple experiments and look at positional specific relationships between sequencing tags, motifs, and other features. You do not need to use the peak finding methods in this package to use motif finding.<br />
<em>Versions Available:</em> 4.7</p>
<p><strong><a href="http://www.htslib.org/" name="htslib">HTSlib</a></strong><br />
A C library for reading/writing high-throughput sequencing data<br />
<em>Versions Available:</em> 1.2.1, 1.3</p>
<p><strong><a href="http://software.broadinstitute.org/software/igv/ " name="igv">IGV</a></strong><br />
The Integrative Genomics Viewer (IGV) is a high-performance visualization tool for interactive exploration of large, integrated genomic datasets.<br />
<em>Versions Available:</em> 2.3.23</p>
<p><strong><a href="http://software.broadinstitute.org/software/igv/igvtools" name="igvtools">igvtools</a></strong><br />
The igvtools utility provides a set of tools for pre-processing data files. File names must contain an accepted file extension, e.g. test-xyz.bam. <br />
<em>Versions Available:</em> 2.3.14, 2.3.67</p>
<p><strong><a href="http://download.java.net" name="java">Java</a></strong><br />
Java technology is the foundation of most networked applications and is used worldwide to develop and deliver mobile and nested applications, games, web content and enterprise software.<br />
<em>Versions Available:</em> openjdk-jdk1.6.0_38, openjdk-jdk1.7.0_60, openjdk-jdk1.8.0_72</p>
<p><strong><a href="http://www.cbcb.umd.edu/software/jellyfish/ " name="jellyfish">JELLYFISH</a></strong><br />
JELLYFISH is a tool for fast, memory-efficient counting of k-mers in DNA. <br />
<em>Versions Available:</em> 2.1.3</p>
<p><strong><a href="https://github.com/ENCODE-DCC/kentUtils" name="kentUtils">kentUtils</a></strong><br />
UCSC command-line bioinformatic utilities, implemented by Jim Kent<br />
<em>Versions Available:</em> 302.1.0</p>
<p><strong><a href="http://kmergenie.bx.psu.edu/ " name="kmergenie">KmerGenie</a></strong><br />
KmerGenie estimates the best k-mer length for genome de novo assembly.  <br />
<em>Versions Available:</em> 1.5692</p>
<p><strong><a href="https://sourceforge.net/p/krona/wiki/KronaTools/ " name="KronaTools">KronaTools</a></strong><br />
Krona Tools is a set of scripts to create Krona charts from several Bioinformatics tools as well as from text and XML files.<br />
<em>Versions Available:</em> 2.6.1</p>
<p><strong><a href="http://www.netlib.org/lapack/ " name="lapack">LAPACK</a></strong><br />
LAPACK provides routines for solving systems of simultaneous linear equations, least-squares solutions of linear systems of equations, eigenvalue problems, and singular value problems.<br />
<em>Versions Available:</em> 3.5.0</p>
<p><strong><a href="http://www.10xgenomics.com/" name="longranger">Long Ranger</a></strong><br />
Long Ranger is a set of analysis pipelines that processes Chromium sequencing output to align reads and call and phase SNPs, indels, and structural variants.<br />
<em>Versions Available:</em> 2.1.2</p>
<p><strong><a href="http://liulab.dfci.harvard.edu/MACS/ " name="MACS">MACS</a></strong><br />
Model-based Analysis of ChIP-Seq (MACS) on short reads sequencers such as Genome Analyzer (Illumina / Solexa)<br />
<em>Versions Available:</em> 2.0.10.09132012</p>
<p><strong><a href="https://pypi.python.org/pypi/MACS2 " name="MACS2">MACS2</a></strong><br />
Novel algorithm, named Model-based Analysis of ChIP-Seq (MACS), for identifying transcript factor binding sites. <br />
<em>Versions Available:</em> 2.1.0.20140616, 2.1.0.20151222, 2.1.1.20160309</p>
<p><strong><a href="https://www.mdc-berlin.de/8551903/en/" name="mirdeep2">miRDeep2</a></strong><br />
miRDeep2 is a completely overhauled tool which discovers microRNA genes by analyzing sequenced RNAs. The tool reports known and hundreds of novel microRNAs with high accuracy in seven species representing the major animal clades. The low consumption of time and memory combined with user-friendly interactive graphic output makes miRDeep2 accessible for straightforward application in current reasearch.<br />
<em>Versions Available:</em> 0.0.8</p>
<p><strong><a href="https://www.mpich.org/ " name="mpich">mpich</a></strong><br />
MPICH is a high performance and widely portable implementation of the Message Passing Interface (MPI) standard.<br />
<em>Versions Available:</em> 3.1.4</p>
<p><strong><a href="https://bitbucket.org/mugqic/mugqic_pipelines" name="mugqic_pipelines">mugqic_pipelines</a></strong><br />
MUGQIC pipelines consist of Python scripts which create a list of jobs running Bash commands. Those scripts support dependencies between jobs and smart restart mechanism if some jobs fail during pipeline execution.  Jobs can be submitted in different ways: by being sent to a PBS scheduler like Torque or by being run as a series of commands in batch through a Bash script<br />
<em>Versions Available:</em> 2.0.1, 2.0.2, 2.1.0, 2.1.1, 2.2.0, 2.2.1</p>
<p><strong><a href="https://bitbucket.org/mugqic/rpackages/src/29b7650e3b38f2f1e16ffb529715268999ec3a14/gqSeqUtils/DESCRIPTION?at=master&amp;fileviewer=file-view-default " name="mugqic_R_packages">mugqic_R_packages</a></strong><br />
This library implements various -seq downstream analysis, as well as Nozzle-based reporting for mugqic_pipelines.<br />
<em>Versions Available:</em> 1.0.1, 1.0.2, 1.0.3, 1.0.4</p>
<p><strong><a href="https://bitbucket.org/mugqic/mugqic_tools" name="mugqic_tools">mugqic_tools</a></strong><br />
Perl, python, R, awk and sh scripts use in several bioinfomatics pipelines of the MUGQIC PIPELINE. <br />
<em>Versions Available:</em> 2.0.2, 2.0.3, 2.1.0, 2.1.1, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.1.7</p>
<p><strong><a href="http://mummer.sourceforge.net/ " name="MUMmer">MUMmer</a></strong><br />
Ultra-fast alignment of large-scale DNA and protein sequences<br />
<em>Versions Available:</em> 3.23</p>
<p><strong><a href="http://www.drive5.com/muscle/downloads.htm" name="MUSCLE">MUSCLE</a></strong><br />
Program for creating multiple alignments of protein sequences.<br />
<em>Versions Available:</em> 3.8.31</p>
<p><strong><a href="http://www.broadinstitute.org/cancer/cga/mutect" name="mutect">MuTect</a></strong><br />
Reliable and accurate identification of somatic point mutations in next generation sequencing data of cancer genomes<br />
<em>Versions Available:</em> 1.1.6</p>
<p><strong><a href="https://github.com/richardmleggett/nextclip/" name="nextclip">NextClip</a></strong><br />
Tool for analysing reads from LMP libraries, generating a comprehensive quality report and extracting good quality trimmed and deduplicated reads<br />
<em>Versions Available:</em> b833dd9</p>
<p><strong><a href="http://www.openblas.net/" name="OpenBLAS">OpenBLAS</a></strong><br />
Optimized BLAS library based on GotoBLAS2 1.13 BSD version<br />
<em>Versions Available:</em> 0.2.14, 0.2.17</p>
<p><strong><a href="http://pandoc.org/" name="pandoc">Pandoc</a></strong><br />
Universal document converter<br />
<em>Versions Available:</em> 1.13.1, 1.15.2</p>
<p><strong><a href="http://www.gnu.org/software/parallel/" name="parallel">parallel</a></strong><br />
Shell tool for executing jobs in parallel using one or more computers<br />
<em>Versions Available:</em> 20130822</p>
<p><strong><a href="http://apps.man.poznan.pl/trac/pbs-drmaa" name="pbs-drmaa">pbs-drmaa</a></strong><br />
DRMAA for Torque/PBS Pro is implementation of Open Grid Forum  DRMAA (Distributed Resource Management Application API) specification for submission and control jobs to PBS systems<br />
<em>Versions Available:</em> 1.0.18</p>
<p><strong><a href="https://www.perl.org/" name="perl">perl</a></strong><br />
Feature-rich programming language<br />
<em>Versions Available:</em> 5.18.2, 5.22.1</p>
<p><strong><a href="https://sourceforge.net/projects/picard/" name="picard">Picard</a></strong><br />
Set of tools (in Java) for working with next generation sequencing data in the BAM format<br />
<em>Versions Available:</em> 1.118, 1.123, 2.0.1</p>
<p><strong><a href="http://zlib.net/pigz/" name="pigz">pigz</a></strong><br />
Replacement for gzip that exploits multiple processors and multiple cores when compressing data<br />
<em>Versions Available:</em> 2.3</p>
<p><strong><a href="http://prinseq.sourceforge.net/" name="prinseq-lite">PRINSEQ-lite</a></strong><br />
Used to filter, reformat, or trim your genomic and metagenomic sequence data<br />
<em>Versions Available:</em> 0.20.3, 0.20.4</p>
<p><strong><a href="http://prodigal.ornl.gov/" name="prodigal">Prodigal</a></strong><br />
Prodigal (Prokaryotic Dynamic Programming Genefinding Algorithm) is a microbial (bacterial and archaeal) gene finding program developed at Oak Ridge National Laboratory and the University of Tennessee.<br />
<em>Versions Available:</em> 2.6.3</p>
<p><strong><a href="https://www.python.org/" name="python">Python</a></strong><br />
Programming language that lets you work quickly and integrate systems more effectively<br />
<em>Versions Available:</em> 2.7.8, 2.7.10_qiime, 2.7.11, 2.7.12, 2.7.13, 3.4.0, 3.5.2</p>
<p><strong><a href="https://bitbucket.org/kokonech/qualimap/downloads" name="qualimap">qualimap</a></strong><br />
Qualimap is a platform-independent application written in Java and R that provides both a Graphical User Interface (GUI) and a command-line interface to facilitate the quality control of alignment sequencing data.<br />
<em>Versions Available:</em> 2.2.1</p>
<p><strong><a href="https://www.bioconductor.org/" name="R_Bioconductor">R_Bioconductor</a></strong><br />
face (GUI) and a command-line interface to facilitate the quality control of alignment sequencing data<br />
<em>Versions Available:</em> 3.1.2_3.0, 3.2.3_3.2</p>
<p><strong><a href="http://denovoassembler.sourceforge.net/" name="ray">Ray</a></strong><br />
Parallel genome assemblies for parallel DNA sequencing<br />
<em>Versions Available:</em> 2.3.1</p>
<p><strong><a href="http://www.cbs.dtu.dk/services/RNAmmer/" name="rnammer">RNAmmer</a></strong><br />
Predicts 5s/8s, 16s/18s, and 23s/28s ribosomal RNA in full genome sequences.<br />
<em>Versions Available:</em> 1.2</p>
<p><strong><a href="https://www.broadinstitute.org/cancer/cga/rna-seqc" name="rnaseqc">RNA-SeQC</a></strong><br />
Java program which computes a series of quality control metrics for RNA-seq data<br />
<em>Versions Available:</em> 1.1.7, 1.1.8</p>
<p><strong><a href="http://deweylab.github.io/RSEM/" name="rsem">RSEM</a></strong><br />
Accurate quantification of gene and isoform expression from RNA-Seq data<br />
<em>Versions Available:</em> 1.2.12</p>
<p><strong><a href="http://www.htslib.org/" name="samtools">SAMtools</a></strong><br />
A suite of programs for interacting with high-throughput sequencing data.<br />
<em>Versions Available:</em> 0.1.19, 1.0, 1.2, 1.3, 1.3.1</p>
<p><strong><a href="http://scalpel.sourceforge.net/" name="scalpel">Scalpel</a></strong><br />
Software package for detecting INDELs (INsertions and DELetions) mutations in a reference genome<br />
<em>Versions Available:</em> 0.3.2, 0.5.2</p>
<p><strong><a href="http://sites.psu.edu/axtell/?s=shortstack" name="ShortStack">ShortStack</a></strong><br />
Tool developed to process and analyze small RNA-seq data with respect to a reference genome, and output a comprehensive and informative annotation of all discovered small RNA genes<br />
<em>Versions Available:</em> 3.3</p>
<p><strong><a href="http://www.cbs.dtu.dk/services/SignalP/" name="signalp">SignalP</a></strong><br />
Predicts the presence and location of signal peptide cleavage sites in amino acid sequences from different organisms<br />
<em>Versions Available:</em> 4.1</p>
<p><strong><a href="https://omictools.com/http://www.pacb.com/products-and-services/analytical-software/smrt-analysis/" name="smrtanalysis">SMRT-Analysis</a></strong><br />
Pacbio secondary analysis through a graphical or command-line user interface.<br />
<em>Versions Available:</em> 2.3.0.140936.p1, 2.3.0.140936.p2, 2.3.0.140936.p4, 2.3.0.140936.p5</p>
<p><strong><a href="http://korflab.ucdavis.edu/software.html" name="snap">SNAP</a></strong><br />
General purpose gene finding program suitable for both eukaryotic and prokaryotic genomes<br />
<em>Versions Available:</em> &#039;2013-11-29&#039;</p>
<p><strong><a href="http://snpeff.sourceforge.net/" name="snpEff">SnpEff</a></strong><br />
Variant annotation and effect prediction tool. It annotates and predicts the effects of variants on genes<br />
<em>Versions Available:</em> 3.6, 4.0, 4.2</p>
<p><strong><a href="http://sphinx-doc.org/" name="sphinx">Sphinx</a></strong><br />
Sphinx is a tool that makes it easy to create intelligent and beautiful documentation of Python projects<br />
<em>Versions Available:</em> master</p>
<p><strong><a href="http://cab.spbu.ru/software/spades/" name="SPAdes">SPAdes</a></strong><br />
SPAdes â€“ St. Petersburg genome assembler â€“ is an assembly toolkit containing various assembly pipelines.<br />
<em>Versions Available:</em> 3.10.0</p>
<p><strong><a href="http://www.raetschlab.org/suppl/spladder" name="SplAdder">SplAdder</a></strong><br />
Splicing Adder, a toolbox for alternative splicing analysis based on RNA-Seq alignment data. Briefly, the software takes a given annotation and RNA-Seq read alignments, transforms the annotation into a splicing graph representation, augments the splicing graph with additional information extracted from the read data, extracts alternative splicing events from the graph and quantifies the events. <br />
<em>Versions Available:</em> 1.0.0</p>
<p><strong><a href="https://github.com/alexdobin/STAR" name="star">STAR</a></strong><br />
Spliced Transcripts Alignment to a Reference. Based on a previously undescribed RNA-seq alignment algorithm that uses sequential maximum mappable seed search in uncompressed suffix arrays followed by seed clustering and stitching procedure.<br />
<em>Versions Available:</em> 2.4.0f1, 2.5.0c, 2.5.1b, 2.5.2a, 2.5.2b</p>
<p><strong><a href="https://support.10xgenomics.com/" name="supernova">Supernova</a></strong><br />
Supernova is a software package for de novo assembly from Chromium Linked-Reads that are made from a single whole-genome library from an individual DNA source.<br />
<em>Versions Available:</em> 1.1.4</p>
<p><strong><a href="http://www.htslib.org/doc/tabix.html" name="tabix">Tabix</a></strong><br />
Tabix indexes a TAB-delimited genome position file in.tab.bgz and creates an index file ( in.tab.bgz.tbi or in.tab.bgz.csi ) when region is absent from the command-line.<br />
<em>Versions Available:</em> 0.2.6</p>
<p><strong><a href="http://www.cbs.dtu.dk/~krogh/TMHMM/" name="tmhmm">TMHMM</a></strong><br />
Predicting Transmembrane Protein Topology with a Hidden Markov Model<br />
<em>Versions Available:</em> 2.0c</p>
<p><strong><a href="https://bitbucket.org/mugqic/mugqic_tools" name="tools">tools</a></strong><br />
Perl, Python, R, awk and sh scripts use in several bioinfomatics pipelines of the MUGQIC PIPELINES repo.<br />
<em>Versions Available:</em> 1.10.4</p>
<p><strong><a href="http://tophat.cbcb.umd.edu" name="tophat">TopHat</a></strong><br />
TopHat is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons. <br />
<em>Versions Available:</em> 2.0.13, 2.0.14</p>
<p><strong><a href="http://transdecoder.github.io" name="TransDecoder">TransDecoder</a></strong><br />
TransDecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks<br />
<em>Versions Available:</em> 2.0.1</p>
<p><strong><a href="http://www.usadellab.org/cms/?page=trimmomatic" name="trimmomatic">Trimmomatic</a></strong><br />
Trimmomatic performs a variety of useful trimming tasks for illumina paired-end and single ended data.The selection of trimming steps and their associated parameters are supplied on the command line.<br />
<em>Versions Available:</em> 0.32, 0.35, 0.36</p>
<p><strong><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki" name="trinity">Trinity</a></strong><br />
Trinity assembles transcript sequences from Illumina RNA-Seq data <br />
<em>Versions Available:</em> 20140413p1, 2.0.4, 2.1.1, 2.2.0</p>
<p><strong><a href="https://trinotate.github.io/" name="trinotate">Tinotate</a></strong><br />
A comprehensive annotation suite for functional annotation of transcriptomes, particularly de novo assembled transcriptomes, from model or non-model organisms. Trinotate makes use of a number of different well referenced methods for functional annotation including homology search to known sequence data (BLAST+/SwissProt), protein domain identification (HMMER/PFAM), protein signal peptide and transmembrane domain prediction (signalP/tmHMM), and leveraging various annotation databases (eggNOG/GO/Kegg databases). <br />
<em>Versions Available:</em> 20131110, 2.0.1, 2.0.2</p>
<p><strong><a href="http://hgdownload.cse.ucsc.edu/admin/jksrc.zip" name="ucsc">UCSC tools</a></strong><br />
UCSC genome browser &#039;kent&#039; bioinformatic utilities<br />
<em>Versions Available:</em> 20140212, v326</p>
<p><strong><a href="http://drive5.com/usearch/" name="usearch">USEARCH</a></strong><br />
Ultra-fast search for high-identity top hit or hits from sequence files<br />
<em>Versions Available:</em> 7.0.1090, 8.1.1861</p>
<p><strong><a href="http://varscan.sourceforge.net/" name="VarScan">VarScan</a></strong><br />
VarScan is a platform-independent mutation caller for targeted, exome, and whole-genome resequencing data generated on Illumina, SOLiD, Life/PGM, Roche/454 and similar instruments. It can be used to detect different types of variation: Germline variants, multi-sample variants, somatic mutations and somatic copy number alterations <br />
<em>Versions Available:</em> 2.3.9</p>
<p><strong><a href="http://vcftools.sourceforge.net/" name="vcftools">VCFtools</a></strong><br />
A program package that can be used to perform the following operations on standard variants (VCF) files: Filter out specific variantsCompare filesSummarize variantsConvert to different file typesValidate and merge filesCreate intersections and subsets of variants<br />
<em>Versions Available:</em> 0.1.11, 0.1.14</p>
<p><strong><a href="http://genome.sph.umich.edu/wiki/VerifyBamID" name="verifyBamID">VerifyBamID</a></strong><br />
Verifies whether the reads in particular file match previously known genotypes for an individual (or group of individuals), and checks whether the reads are contaminated as a mixture of two samples. verifyBamID can detect sample contamination and swaps when external genotypes are available. When external genotypes are not available, verifyBamID still robustly detects sample swaps<br />
<em>Versions Available:</em> devMaster_20151216</p>
<p><strong><a href="https://www.tbi.univie.ac.at/RNA/" name="ViennaRNA">ViennaRNA</a></strong><br />
The ViennaRNA Package consists of a C code library and several stand-alone programs for the prediction and comparison of RNA secondary structures.<br />
<em>Versions Available:</em> 2.3.0</p>
<p><strong><a href="https://github.com/torognes/vsearch" name="vsearch">VSEARCH</a></strong><br />
VSEARCH supports de novo and reference based chimera detection, clustering, full-length and prefix dereplication, reverse complementation, masking, all-vs-all pairwise global alignment, exact and global alignment searching, shuffling, subsampling and sorting. It also supports FASTQ file analysis, filtering and conversion.<br />
<em>Versions Available:</em> 1.11.1</p>
<p><strong><a href="http://genome.sph.umich.edu/wiki/vt" name="vt">vt</a></strong><br />
A tool set for short variant discovery in genetic sequence data.<br />
<em>Versions Available:</em> 0.57</p>
<p><strong><a href="http://weblogo.threeplusone.com/" name="weblogo">WebLogo</a></strong><br />
A tool for creating sequence logos from biological sequence alignments.  It can be run on the command line as a standalone webserver, as a CGI webapp, or as a python library.<br />
<em>Versions Available:</em> 2.8.2, 3.3</p>
<p><strong><a href="http://wgs-assembler.sourceforge.net" name="wgs-assembler">Celera Assembler</a></strong><br />
A de novo whole-genome shotgun (WGS) DNA sequence assembler. It reconstructs long sequences of genomic DNA from fragmentary data produced by whole-genome shotgun sequencing<br />
<em>Versions Available:</em> 8.1, 8.2, 0.0, 0.2</p>
<p></p>

			</div> <!-- .et_pb_code -->
			</div> <!-- .et_pb_column -->
					
			</div> <!-- .et_pb_row -->
				
			</div> <!-- .et_pb_section -->
					</div> <!-- .entry-content -->

				
				</article> <!-- .et_pb_post -->

			

</div> <!-- #main-content -->


	<span class="et_pb_scroll_top et-pb-icon"></span>


			<footer id="main-footer">
				
<div class="container">
	<div id="footer-widgets" class="clearfix">
	<div class="footer-widget"><div id="text-3" class="fwidget et_pb_widget widget_text"><h4 class="title">About</h4>			<div class="textwidget"><hr style="border:1px thin #FFFFFF;"</hr>
<p class="footerblurb" style="font-size:0.9em">C3G provides bioinformatics analysis and HPC services for the life science research community. It  has created a distributed innovation node with broad expertise in bioinformatics, providing customized and case-by-case analysis services as well as an extensive suite of software solutions for the genomics community. </br><a href="http://www.computationalgenomics.ca/about/" style="color:#4397df;">Learn more.</a></p>
</div>
		</div> <!-- end .fwidget --></div> <!-- end .footer-widget --><div class="footer-widget"><div id="text-4" class="fwidget et_pb_widget widget_text"><h4 class="title">Services</h4>			<div class="textwidget"><hr style="border:1px thin #FFFFFF;">

<div style="margin-top:15px; width:100%!important" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/services/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff;"></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/services/">Analysis Services</a>
</h4>
</div>
</div>
</div>

<div style="margin-top:15px;" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/tools/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff;"></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/tools/">Bioinformatic Tools</a>
</h4>
</div>
</div>
</div>

<div  style="margin-top:15px;" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/2018-2/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff;"></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/2019-2/">Training & Workshops</a>
</h4>
</div>
</div>
</div>
<div width="100%"></div></div>
		</div> <!-- end .fwidget --></div> <!-- end .footer-widget --><div class="footer-widget"><div id="text-8" class="fwidget et_pb_widget widget_text"><h4 class="title">Contact</h4>			<div class="textwidget"><hr style="border:1px thin #FFFFFF;"</hr>
<div style="margin-top:15px;" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/montreal-node/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff; font-size:16px;" ></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/montreal-node/">Montreal Node</a>
</h4>
</div>
</div>
</div>


<div style="margin-top:15px;" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/toronto-node/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff; font-size:16px;" ></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/toronto-node/">Toronto Node</a>
</h4>
</div>
</div>
</div>

<div style="margin-top:15px;" class="footerblurb et_pb_blurb et_pb_module et_pb_bg_layout_dark et_pb_text_align_left et_pb_blurb_6 et_pb_blurb_position_left">
<div class="et_pb_blurb_content">
<div class="et_pb_main_blurb_image" style="height:inherit !important; padding-top:inherit !important;">
<a href="http://www.computationalgenomics.ca/contact/">
<span class="et-pb-icon et-waypoint et_pb_animation_top et-animated" style="color: #ffffff; font-size:16px;" ></span>
</a>
</div>
<div class="footerblurb et_pb_blurb_container" style="padding-top:6px">
<h4>
<a href="http://www.computationalgenomics.ca/contact/">info@computationalgenomics.ca</a>
</h4>
</div>
</div>
</div>
</div>
		</div> <!-- end .fwidget --></div> <!-- end .footer-widget -->	</div> <!-- #footer-widgets -->
</div>	<!-- .container -->

		
				<div id="footer-bottom">
					<div class="container clearfix">
				<ul class="et-social-icons">

	<li class="et-social-icon et-social-twitter">
		<a href="https://twitter.com/C3Genomics" class="icon">
			<span>Twitter</span>
		</a>
	</li>
	<li class="et-social-icon et-social-google-plus">
		<a href="https://groups.google.com/forum/#!forum/genpipes" class="icon">
			<span>Google</span>
		</a>
	</li>

</ul>
						<p id="footer-info">Copyright © 2020 C3G. All rights reserved.</p>
					</div>	<!-- .container -->
				</div>
			</footer> <!-- #main-footer -->
		</div> <!-- #et-main-area -->


	</div> <!-- #page-container -->

	<style type="text/css" id="et-builder-advanced-style">
				
.et_pb_fullwidth_header_0.et_pb_fullwidth_header { background-image: url(http://www.computationalgenomics.ca/wp-content/uploads/2016/02/stem-cell-test2.jpg); }
			</style><style type="text/css" id="et-builder-page-custom-style">
				 .et_pb_bg_layout_dark { color: #ffffff !important; } .page.et_pb_pagebuilder_layout #main-content { background-color: rgba(255,255,255,0); } .et_pb_section { background-color: #ffffff; }
			</style><script type='text/javascript'>
/* <![CDATA[ */
var wpcf7 = {"apiSettings":{"root":"http:\/\/www.computationalgenomics.ca\/wp-json\/","namespace":"contact-form-7\/v1"},"recaptcha":{"messages":{"empty":"Please verify that you are not a robot."}}};
/* ]]> */
</script>
<script type='text/javascript' src='http://www.computationalgenomics.ca/wp-content/plugins/bwp-minify/min/?f=wp-content/themes/Divi/includes/builder/scripts/frontend-builder-global-functions.js,wp-content/plugins/display-medium-posts/public/js/display-medium-posts-public.js,wp-content/plugins/jquery-validation-for-contact-form-7/js/jquery.validate.min.js,wp-content/plugins/jquery-validation-for-contact-form-7/js/jquery.jvcf7_validation.js,wp-content/themes/Divi-child/js/json_downloader.js,wp-content/themes/Divi/includes/builder/scripts/jquery.mobile.custom.min.js,wp-content/themes/Divi/js/custom.js,wp-content/themes/Divi/js/smoothscroll.js,wp-content/themes/Divi/includes/builder/scripts/jquery.fitvids.js,wp-content/themes/Divi/includes/builder/scripts/waypoints.min.js'></script>
<script type='text/javascript'>
/* <![CDATA[ */
var et_pb_custom = {"ajaxurl":"http:\/\/www.computationalgenomics.ca\/wp-admin\/admin-ajax.php","images_uri":"http:\/\/www.computationalgenomics.ca\/wp-content\/themes\/Divi\/images","builder_images_uri":"http:\/\/www.computationalgenomics.ca\/wp-content\/themes\/Divi\/includes\/builder\/images","et_frontend_nonce":"7566a68bb9","subscription_failed":"Please, check the fields below to make sure you entered the correct information.","et_ab_log_nonce":"3b9076aacd","fill_message":"Please, fill in the following fields:","contact_error_message":"Please, fix the following errors:","invalid":"Invalid email","captcha":"Captcha","prev":"Prev","previous":"Previous","next":"Next","wrong_captcha":"You entered the wrong number in captcha.","is_builder_plugin_used":"","is_divi_theme_used":"1","widget_search_selector":".widget_search","is_ab_testing_active":"","page_id":"2713","unique_test_id":"","ab_bounce_rate":"5","is_cache_plugin_active":"no","is_shortcode_tracking":""};
/* ]]> */
</script>
<script type='text/javascript' src='http://www.computationalgenomics.ca/wp-content/plugins/bwp-minify/min/?f=wp-content/themes/Divi/includes/builder/scripts/jquery.magnific-popup.js,wp-content/themes/Divi/includes/builder/scripts/frontend-builder-scripts.js,wp-content/plugins/divi-builder/core/admin/js/common.js,wp-includes/js/wp-embed.min.js'></script>
</body>
</html>